## Assignment 2 - Data Stewardship

## Deadline:  Nov 28, 23:59,  [y.he2@uva.nl](mailto:y.he2@uva.nl)

## Upload your notebook file on Canvas

## Notebook made by  

|** Name** | **Student id** | **email**|
|:- |:-|:-|
|. | | |
|  | |. |

 We (enlisted above) declare that:
1. Our assignment will be our own work.
2. We shall not make solutions to the assignment available to anyone else.
3. We shall not engage in any other activities that will dishonestly improve my results or dishonestly improve or hurt the results of others.

Note: 
- your notebook should give no errors when executed as "Run All" cells
- ** Use markdown cell setting for comments/additional explanation **

In [1]:
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as stats
from pandas import Series, DataFrame
import pandas as pd
plt.rc('figure', figsize=(10, 6))
%matplotlib inline
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [2]:
df=pd.read_csv('ClothingStore.csv')

# <font color='blue'>Introduction</font>

In this assignment, you are acting as analysts for a retail clothing store chain. 

The *ClothingStore* data set represents actual data provided by a clothing store chain in New England.

Our data mining task is to classify which customers will respond to a direct-mail marketing promotion, based on information collected about the customers.

You become more familiar with the data set using EDA, graphical and descriptive statistical methods for learning about data.

# <font color='blue'>Getting a feel for the data set</font>

In [3]:
df.RESP.value_counts(normalize=True)
# dispaly the proportions of the unique values in RESP

0    0.834647
1    0.165353
Name: RESP, dtype: float64

The target variable will be <font color='blue'>RESP</font>. As can be seen above, the overall proportion of responders to the direct-mail marketing promotion is 16.54%.

# Question 1
One of the variable, <font color='blue'>CLUSTYPE</font>, contains the market segmentation category for each customer as defined by Nielsen Claritas.

- (a) Generate a table that contains the top 5 cluster type with the highest frequencies (so most prevalent lifestyle cluster type).
- (b) Match the cluster type in the file *Lifestyle.htm* and report the labels. 

    Hint: use ``pd.read_html('Lifestyle.htm')[0]`` to load the htm-file, and use ``DataFrame.set_index()`` method to set the DataFrame index using existing columns.

# <font color='blue'>Negative Amounts Spent?</font>

# Question 2
For many of the amounts-spent fields and the proportions-spent fields, some of the customers have negative values for the amount or proportion of money spent. How can this be? 

Now, the data were collected within a particular time period, which is unspecified, perhaps a month or a quarter. It is possible for a customer to have bought some clothing in a prior period, and returned the purchased clothing in the time period from which the data are collected. If this customer also did not make any major purchases during the time period of interest, then the net sales for this customer would be negative.

- (a) For the variable <font color='blue'>PSSPEND</font> (amount spend at the PS franchise), determine the number customers with a negative amount spent.
- (b) Generate a flag variable that recodes <font color='blue'>PSSPEND</font> in 'Neg','Pos' and 'Zero'.
- (c) Using this flag variable generate a crosstab showing the response *rate* for the three spending  categories 'Neg','Pos' and 'Zero'.
- (d) In the remainder of this assignment, we consider the absolute value of spending observations. Motivate why this is a reasonable choice given the results of 2(c).

# <font color='blue'>Transformations to Achieve Normality or Symmetry</font>

# Question 3
Most of the numeric fields are right-skewed. For example, the column <font color='blue'>HI</font> contains a variable which takes large values for customers who purchase only a few different classes of clothes (e.g., blouses, legwear, pants), and small values for customers who purchase many different classes of clothes (hence giving an indication of the product uniformity).
- (a) Generate a histogram of <font color='blue'>HI</font> using 50 bins. Do you see some spikes?

Many data mining methods and models function best when the variables are normally distributed, or,
at least when they are symmetric. The analyst may choose transformations such as the natural log (<tt>np.log</tt>) transformation or the square root transformation (<tt>np.sqrt</tt>).
- (b) Transform <font color='blue'>HI</font> in <font color='blue'>HI_log</font> using the log transformation and generate again a histogram. Compare the moment skewness of <font color='blue'>HI</font> and <font color='blue'>HI_log</font> using the <tt>DataFrame</tt>.**skew()** method.
- (c) The column <font color='blue'>PBLOUSES</font> contains information about the percentage spent on blouses. This could be negative, so first define <font color='blue'>PBLOUSES_abs</font> as the absolute value of <font color='blue'>PBLOUSES</font>. Do you find any difficulties in the log transformation? Motivate your answer.

# <font color='blue'>Exploring the relationship between the predictors and the response</font>

# Question 4
Next, we would like to investigate the relationship with the following flag variable and the response rate: 

- <font color='blue'>WEB</font>: equals 1 for those who have made purchases via the company’s web shopping option

Please, answer the following three questions:

- (a) Using crosstabs determine how many times Web buyers are more likely to respond to those have not made a purchase via the Web.

- (b) Using normalized bar chart with the response (<font color='blue'>RESP</font>) overlay, describe for <font color='blue'>WEB</font> the relationship with the target variable.

- (c) Using <tt>stats.chi2_contingency()</tt> test if the relationship between <font color='blue'>WEB</font> and <font color='blue'>RESP</font> is significant at a 5% nominal significance level. State the null hypothese, p-value, and decision.